sonar-scanner-engine/src/main/java/org/sonar/batch/bootstrapper/Batch.java

```diff
@@ -67,16 +67,26 @@ public LoggingConfiguration getLoggingConfiguration() {
   public synchronized Batch execute() {
     configureLogging();
     doStart();
-    boolean threw = true;
-    try {
-      doExecuteTask(globalProperties);
-      threw = false;
-    } finally {
-      doStop(threw);
+    try (Foo foo = new Foo() {
+      @Override
+      public void run() {
+        doExecuteTask(globalProperties);
+      }
+
+      @Override
+      public void close() {
+        doStop();
+      }
+    }) {
+      foo.run();
     }
     return this;
   }
 
+  public abstract class Foo implements Runnable, AutoCloseable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)